pkg/controllers/workapplier/controller.go

```diff
@@ -71,17 +77,136 @@ const (
 	workFieldManagerName = "work-api-agent"
 )
 
+var (
+	workAgeToReconcile = 1 * time.Hour
+)
+
+// Custom type to hold a reconcile.Request and a priority value
+type priorityQueueItem struct {
+	reconcile.Request
+	Priority int
+}
+
+// PriorityQueueEventHandler is a custom event handler for adding objects to the priority queue.
+type PriorityQueueEventHandler struct {
+	Queue  priorityqueue.PriorityQueue[priorityQueueItem] // The priority queue to manage events
+	Client client.Client                                  // store the client to make API calls
+}
+
+// Implement priorityqueue.Item interface for priorityQueueItem
+func (i priorityQueueItem) GetPriority() int {
+	return i.Priority
+}
+
+func (h *PriorityQueueEventHandler) WorkPendingApply(ctx context.Context, obj client.Object) bool {
+	var work fleetv1beta1.Work
+	ns := obj.GetNamespace()
+	name := obj.GetName()
+	err := h.Client.Get(ctx, client.ObjectKey{
+		Namespace: ns,
+		Name:      name,
+	}, &work)
+	if err != nil {
+		// Log and return
+		klog.ErrorS(err, "Failed to get the work", "name", name, "ns", ns)
+		return true
+	}
+	availCond := meta.FindStatusCondition(work.Status.Conditions, fleetv1beta1.WorkConditionTypeAvailable)
+	appliedCond := meta.FindStatusCondition(work.Status.Conditions, fleetv1beta1.WorkConditionTypeApplied)
+
+	// check if the object has been recently modified
+	availCondLastUpdatedTime := availCond.LastTransitionTime.Time
+	appliedCondLastUpdatedTime := appliedCond.LastTransitionTime.Time
+	if time.Since(availCondLastUpdatedTime) < workAgeToReconcile || time.Since(appliedCondLastUpdatedTime) < workAgeToReconcile {
+		return true
+	}
+
+	if condition.IsConditionStatusTrue(availCond, work.GetGeneration()) &&
+		condition.IsConditionStatusTrue(appliedCond, work.GetGeneration()) {
+		return false
+	}
+
+	// Work not yet applied
+	return true
+}
+
+func (h *PriorityQueueEventHandler) AddToPriorityQueue(ctx context.Context, obj client.Object, alwaysAdd bool) {
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: obj.GetNamespace(),
+			Name:      obj.GetName(),
+		},
+	}
+
+	objAge := time.Since(obj.GetCreationTimestamp().Time)
+
+	var objPriority int
+	if alwaysAdd || objAge < workAgeToReconcile || h.WorkPendingApply(ctx, obj) {
```

Comment:
"workAgeToReconcile" is already taken into consideration in the WorkPendingApply function, does it make sense to use it here again?